src/azure/cli/_profile.py

```diff
@@ -334,9 +331,7 @@ def persist_cached_creds(self):
 
             all_creds.extend(self._service_principal_creds)
             cred_file.write(json.dumps(all_creds))
-
-        if new_file:
-            os.chmod(self._token_file, 0o600)
+            cred_file.truncate()#to avoid reopening and write
```

Comment:
no, this is working code. and a little surprising to me. Checkout this [SO thread](http://stackoverflow.com/questions/2424000/read-and-overwrite-a-file-in-python)